src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```
@@ -2116,35 +2123,53 @@ because xbuild doesn't support framework reference assemblies.
     ResolvedAssemblies="@(_ResolvedAssemblies)"
     ResolvedUserAssemblies="@(_ResolvedUserMonoAndroidAssemblies)"
     ErrorOnCustomJavaObject="$(AndroidErrorOnCustomJavaObject)"
-	ManifestTemplate="$(_AndroidManifestAbs)"
-	MergedManifestDocuments="@(ExtractedManifestDocuments)"
-	Debug="$(AndroidIncludeDebugSymbols)"
-	MultiDex="$(AndroidEnableMultiDex)"
-	NeedsInternet="$(AndroidNeedsInternetPermission)"
-	InstantRunEnabled="$(_InstantRunEnabled)"
-	AndroidSdkPlatform="$(_AndroidApiLevel)"
-	AndroidSdkDir="$(_AndroidSdkDirectory)"
-	PackageName="$(_AndroidPackage)"
-	ManifestPlaceholders="$(AndroidManifestPlaceholders)"
-	OutputDirectory="$(IntermediateOutputPath)android"
-	MergedAndroidManifestOutput="$(IntermediateOutputPath)android\AndroidManifest.xml"
+    ManifestTemplate="$(_AndroidManifestAbs)"
+    Debug="$(AndroidIncludeDebugSymbols)"
+    MultiDex="$(AndroidEnableMultiDex)"
+    NeedsInternet="$(AndroidNeedsInternetPermission)"
+    InstantRunEnabled="$(_InstantRunEnabled)"
+    AndroidSdkPlatform="$(_AndroidApiLevel)"
+    AndroidSdkDir="$(_AndroidSdkDirectory)"
+    PackageName="$(_AndroidPackage)"
+    ManifestPlaceholders="$(AndroidManifestPlaceholders)"
+    OutputDirectory="$(IntermediateOutputPath)android"
+    MergedAndroidManifestOutput="$(IntermediateOutputPath)AndroidManifest.xml"
     UseSharedRuntime="$(AndroidUseSharedRuntime)"
-	EmbedAssemblies="$(EmbedAssembliesIntoApk)"
-	BundledWearApplicationName="$(BundledWearApplicationPackageName)"
-	PackageNamingPolicy="$(AndroidPackageNamingPolicy)"
-	ApplicationJavaClass="$(AndroidApplicationJavaClass)"
-	FrameworkDirectories="$(_XATargetFrameworkDirectories);$(_XATargetFrameworkDirectories)Facades"
-	AcwMapFile="$(_AcwMapFile)"
-	SupportedAbis="@(_BuildTargetAbis)">
+    EmbedAssemblies="$(EmbedAssembliesIntoApk)"
+    BundledWearApplicationName="$(BundledWearApplicationPackageName)"
+    PackageNamingPolicy="$(AndroidPackageNamingPolicy)"
+    ApplicationJavaClass="$(AndroidApplicationJavaClass)"
+    FrameworkDirectories="$(_XATargetFrameworkDirectories);$(_XATargetFrameworkDirectories)Facades"
+    AcwMapFile="$(_AcwMapFile)"
+    SupportedAbis="@(_BuildTargetAbis)">
   </GenerateJavaStubs>
 
   <ItemGroup>
     <FileWrites Include="@(_TypeMapAssemblySource)" />
+    <FileWrites Include="$(IntermediateOutputPath)AndroidManifest.xml" />
   </ItemGroup>
 
   <Touch Files="$(_AndroidStampDirectory)_GenerateJavaStubs.stamp" AlwaysCreate="True" />
 </Target>
 
+<Target Name="_ManifestMerger"
+    Inputs="$(IntermediateOutputPath)AndroidManifest.xml;@(ExtractedManifestDocuments);$(_AndroidBuildPropertiesCache)"
```

Maybe this needs `$(MSBuildAllProjects)` to match `_GenerateJavaStubs`:
```suggestion
    Inputs="$(IntermediateOutputPath)AndroidManifest.xml;@(ExtractedManifestDocuments);$(_AndroidBuildPropertiesCache);$(MSBuildAllProjects)"
```
I could see a `@(Reference)` or `@(PackageReference)` being removed and it wouldn't retrigger this target. You could end up with a removed library's manifest included.

---

It would be OK to bump this time, if that is the case: https://github.com/xamarin/xamarin-android/blob/7ef398af1e23e280070300d430c8610ad25593b7/tests/msbuild-times-reference/MSBuildDeviceIntegration.csv#L10

---

In which case the `ExtractedManifestDocuments` ItemGroup should be updated.

---

But now I think about it, you make a good point. The Documents themselves won't be... 🤔